### PR TITLE
Quantity input widget

### PR DESCRIPTION
**Pull request by spireken**
I put a Scratch build up that shows two buttons, one that makes use of the 'streaming' value changed signal, and the other that uses the committed signal. There's also a button that calls set_value() on the quantity inputs.

### CODE REVIEW

File: Applications/Spire/Include/Spire/Ui/InputWidget.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#ifndef SPIRE_INPUT_WIDGET_HPP
+#define SPIRE_INPUT_WIDGET_HPP
+#include <QLineEdit>
+
+namespace Spire {
+
+  //! Displays a line edit with the Spire style.
+  class InputWidget : public QLineEdit {
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/InputWidget.hpp]: This name is too vague.

File: Applications/Spire/Source/Ui/InputWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#include "Spire/Ui/InputWidget.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace Spire;
+
+InputWidget::InputWidget(QWidget* parent)
+    : QLineEdit(parent) {
+  setStyleSheet(QString(R"(
```
Comment (kranar) [inline Applications/Spire/Source/Ui/InputWidget.cpp]: Not enough functionality here to justify a class. If you want you can write a function that applies a certain style to a class.

File: Applications/Spire/Include/Spire/Ui/QuantityInputWidget.hpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#ifndef SPIRE_NUMBER_INPUT_WIDGET_HPP
+#define SPIRE_NUMBER_INPUT_WIDGET_HPP
+#include <QWidget>
+#include "Nexus/Definitions/Quantity.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/InputWidget.hpp"
+
+namespace Spire {
+
+  //! Displays an input box that accepts integer values.
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/QuantityInputWidget.hpp]: ```Quantity```'s are not integers.

File: Applications/Spire/Source/Ui/InputWidget.cpp (hunk 2 above)
Comment (spireken) [reply Applications/Spire/Source/Ui/InputWidget.cpp]: So like:

// Ui.cpp

apply_line_edit_style(QWidget* widget) {
    static auto style = ...;
    widget->setStylesheet(style);
}

File: Applications/Spire/Include/Spire/Ui/QuantityInputWidget.hpp [hunk 4]
```diff
@@ -1,14 +1,13 @@
 #ifndef SPIRE_NUMBER_INPUT_WIDGET_HPP
 #define SPIRE_NUMBER_INPUT_WIDGET_HPP
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/QuantityInputWidget.hpp]: Fix the include guard.

**Comment by spireken** [comment]
Has this been merged?

**Pull request merged**